File: 백준/Silver/11723. 집합/집합.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int m, s; cin >> m; s = 0;
    while(m--){
        string cmd; cin >> cmd;
        if(cmd == "all") s = (1<<21)-1; 
        else if(cmd == "empty") s = 0;
        else{
            int x; cin >> x;
            if(cmd == "add") s |= (1<<x);
            else if(cmd == "remove") s &= ~(1<<x);
            else if(cmd == "check") cout << (s & (1<<x) ? 1 : 0) << '\n';
            else if(cmd == "toggle") s ^= (1<<x);
        }
    }
    return 0;
}

File: 백준/Silver/11723. 집합/집합.py
import sys
def print(*args, sep=" ", end="\n"): return sys.stdout.write(sep.join(map(str, args)) + end)
def input(): return sys.stdin.readline().rstrip()

def main():
    s = set()
    m = int(input())
    for _ in range(m):
        cmd = input().split()
        if len(cmd) == 2: x = int(cmd[1])
        if cmd[0] == "add": s.add(x)
        elif cmd[0] == "remove": s.discard(x)
        elif cmd[0] == "check":
            if x in s: print(1)
            else: print(0)
        elif cmd[0] == "toggle":
            if x in s: s.discard(x)
            else: s.add(x)
        elif cmd[0] == "all": s = set(range(1, 21))
        elif cmd[0] == "empty": s.clear()
    return 0
if __name__ == '__main__':
    sys.exit(main())